docs/design_docs/cached_outputs.rst

```diff
@@ -0,0 +1,343 @@
+:ref:`des_unique`) that maps between output files in the work directory and
+those on the LCRC server.  For example:
+
+.. code-block:: bash
+
+    compass cache \
+        ocean/global_ocean/QU240/mesh/mesh \
+        ocean/global_ocean/QU240/PHC/init/initial_state
+
+.. _des_unique:
+
+Design: unique identifier for cached outputs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+Each cached file on the LCRC server will include a date stamp in the file name.
+For example, ``culled_mesh.nc`` will become ``culled_mesh.20210730.nc`` on the
+server.  When ``compass cache`` is called (see :ref:`des_update`), the date
+stamp will default to the date that the call is being made but can be
+overridden with a flag (e.g. ``--date 20210730``).
+
+Each MPAS core in ``compass`` will optionally include a file
+``cached_files.json`` that contains a python dictionary mapping between the
+names of output files in the work directory and those in the ``compass_cache``
+database for that MPAS core on the LCRC server.  For example:
+
+.. code-block:: json
+
+    {
+        "ocean/global_ocean/QU240/mesh/mesh/culled_mesh.nc": "global_ocean/QU240/mesh/mesh/culled_mesh.210727.nc",
+        "ocean/global_ocean/QU240/mesh/mesh/culled_graph.info": "global_ocean/QU240/mesh/mesh/culled_graph.210727.info",
+        "ocean/global_ocean/QU240/mesh/mesh/critical_passages_mask_final.nc": "global_ocean/QU240/mesh/mesh/critical_passages_mask_final.210727.nc",
+        "ocean/global_ocean/QU240/PHC/init/initial_state/initial_state.nc": "global_ocean/QU240/PHC/init/initial_state/initial_state.210727.nc",
+        "ocean/global_ocean/QU240/PHC/init/initial_state/init_mode_forcing_data.nc": "global_ocean/QU240/PHC/init/initial_state/init_mode_forcing_data.210727.nc"
+    }
+
+.. _des_normal_or_cached:
+
+Design: either "normal" or "cached" versions of a step
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/07/30
+
+Contributors: Xylar Asay-Davis
+
+A prototype implementation of output caching had separate versions of test
+cases that included cached outputs or depended on earlier test cases with
+cached outputs.  This approach turned out to be very cumbersome.  It added
+many "new" test cases with unique subdirectories in the work directory and
+required predetermining which steps should allow caching.  But this approach
+*did* allow a test suite to include a "normal" version of a step and a "cached"
+version of that same step in the same work directory (and therefore in the same
+test suite).
+
+The proposed design, described in the previous sections, would allow far more
+flexibility about which steps are cached and which are not.  It is not clear
+to me how we achieve this flexibility without requiring that a given step
+either be set up as "normal" or "cached", and not both in the same work
+directory.
+
+
+
+
+Implementation
+--------------
```

Comment:
Waiting until we iterate on requirements and design before I rework the implementation in #184